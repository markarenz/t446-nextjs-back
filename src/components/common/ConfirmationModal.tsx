import styles from '../../styles/modules/ConfirmationModal.module.scss';
import Button from './Button';

type Props = {
  handleOk: Function;
  handleCancel: Function;
  children: JSX.Element;
  title: string;
};

const ConfirmationModal: React.FC<Props> = ({
  handleOk,
  handleCancel,
  children,
  title
}) => {
  return (
    <div className={styles.modalRoot}>
      <div className="card">
        <div className="card-header">{title}</div>
        <div className="card-body">{children}</div>
        <div className="card-actions">
          <span className="mr-1">
            <Button
              disabled={false}
              variant="outlined"
              onClick={handleCancel}
              tabIndex={1}
            >
              <span>Cancel</span>
            </Button>
          </span>
          <span>
            <Button
              disabled={false}
              variant="contained"
              color="primary"
              onClick={handleOk}
              tabIndex={2}
            >
              <span>OK</span>
            </Button>
          </span>
        </div>
      </div>
    </div>
  );
};

export default ConfirmationModal;
