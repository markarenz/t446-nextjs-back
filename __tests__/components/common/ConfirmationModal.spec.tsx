import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import ConfirmationModal from '../../../src/components/common/ConfirmationModal';

const defaultProps = {
  handleOk: jest.fn(),
  handleCancel: jest.fn(),
  title: 'Test Modal'
};

describe('ConfirmationModal', () => {
  it('renders modal', async () => {
    render(
      <ConfirmationModal {...defaultProps}>
        <span>Test</span>
      </ConfirmationModal>
    );
    const element = await screen.findByText('Test');
    expect(element).toBeInTheDocument();
  });
});
