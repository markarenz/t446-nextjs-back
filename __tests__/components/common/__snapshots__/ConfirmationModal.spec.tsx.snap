// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationModal renders modal 1`] = `
<span>
  Test
</span>
`;
